### ISSUE TITLE
Text Alignment

### ISSUE THREAD

**Issue by sroddy** [flutter/flutter_markdown#19]
Hi,
Is there a way to set text alignment properties in the style? I think it would be an important feature in order to properly style the text (e.g. justify text in `p`, center it in `h1/h2`, etc).

Thanks

**Comment by DaveShuckerow** [comment]
Hi Stefano, thanks for reaching out!

Text alignment is a separate property from Text styling in Flutter.  In general, Markdown that I've seen doesn't give much control over alignment: it all tends to be aligned to the start of a line.

It's *possible* to expand the MarkdownStyleSheet to provide alignment as well, but there are a lot of degenerate edge cases you could create that we have to solve for.  Off the top of my head:

 - What if I want to **center strong** text and align regular text left?
 - What if I want to right-align or center list items?  How does the starting bullet point get laid out?

If you need to do this and you're interested in adding support for it, I'm happy to read a PR.  If you do, be sure to explain why the change is useful, and be thorough in testing for edge cases like the two I listed above.

--------

Also, if you're interested in rendering HTML in Flutter, we have an HTML Widget internal to Google that we hope to release to GitHub soon.

**Comment by sroddy** [comment]
In general I was looking for a way of rendering paragraphs of text with some formatting and link handlers, which is a bit simpler than the Flutter's Text/TextSpan API, so I will be really interested in an HTML Widget, especially if it is a bit more customisable but simple enough to use.

About this Widget, I perfectly understand your concerns and I think that a possible approach would be to differentiate between the tags that are `inline` in html and the ones that are naturally `blocks`.
In order not to break BC we could just add optional `ParagraphStyle` attributes (or just the text-align) in the `MarkdownStyleSheet` constructors and utility functions just for these `block` tags (e.g. `pParagraphStyle`, `h1ParagraphStyle`, etc.).

What do you think about that?

**Comment by DaveShuckerow** [comment]
I don't want to introduce another layer of indirection beyond what's already available from Flutter.  

If you want to see what this looks like, my suggestion is to put together a PR adding support for just `headerTextAlign` into the MarkdownStyleSheet.  At that point, I'm happy to read it and we can discuss how to optimize it and what other cases to expand to.  At the present, my understanding is that the cases where changing the text alignment is useful are the case of headers and the case of paragraphs.

Also note, if this looks like it'll add more than a small amount of complexity to the API, I might ask to wait on merging this PR with head until other people come forward with interest in this feature.

**Comment by lemonboston** [comment]
@DaveShuckerow It's nice to hear that there is a HTML Widget in line. Do you happen to have any update on that or could you possibly point me to somewhere? 

We've just started a project in Flutter, it's a rewrite from Android to support iOS, tool. And there are some HTML contents that we will now convert to markdown on the server because of the lack of support for HTML in Flutter, and then use your MarkdownWidget. The question of how to align the text comes up for us as well - that's how I stumbled across this ticket. (We may manage without it, I am just exploring the options currently.)

**Comment by IchordeDionysos** [comment]
For anyone interested I have implemented it now completely as we needed this at simpleclub.

You can use it by adding `flutter_markdown` in the `pubspec.yaml` like this:

```yaml
flutter_markdown:
    git:
        url: git@github.com:simpleclub/flutter_markdown.git
        ref: alignment
```

**Comment by maks** [comment]
@IchordeDionysos thanks for building this! For me your textAlign property was exactly what I needed.

@DaveShuckerow for me the use case is that I'm using `MarkdownBody` as a way of rendering some simple markup (eg. bold text) within 1 or more places in a whole sentence of text sourced from localised  strings, so using `RichText` would be painful BUT I needed to have it center aligned, so using `textAlign: WrapAlignment.center` in `MarkdownStyleSheet` as introduced in @IchordeDionysos fork did exactly what I needed. Since I'm using `MarkdownBody` as essentially a formatted alternative of `Text` widget, it would good if at least the `textAlign` property could be supported for this use case.

**Comment by codegrue** [comment]
I vote to merge this in, or similar functionality.

**Comment by esDotDev** [comment]
Agreed, while it might not technically be in the spec, it's the only thing holding this back as a drop-in replacement for RichTextSpan

**Comment by esDotDev** [comment]
Agreed, while it might not technically be in the spec, it's the only thing holding this back as a drop-in replacement for RichTextSpan. 

Maybe an ExtendedMarkdown widget to be explicit?

**Comment by maks** [comment]
@esDotDev this issue is **not** about specifing text alignment within Markdown text, it's about specifing the text alignment of **rendered** output via the existing style mechanism.

**Comment by maks** [comment]
Actually, @domesticmouse I think perhaps this issue can be closed as #105 was merged a while ago and it added alignment for quite a lot of output elements.

**Comment by domesticmouse** [comment]
Thanks @maks, closing
